src/misc/cross.cpp

```diff
@@ -439,6 +440,224 @@ FILE *fopen_wrap_ro_fallback(const std::string &filename, bool &is_readonly)
 	return fp;
 }
 
+bool wild_match(const char *haystack, char *needle)
+{
+	size_t max, i;
+	for (; *needle != '\0'; ++needle) {
+		switch (*needle) {
+		case '?':
+			if (*haystack == '\0')
+				return false;
+			++haystack;
+			break;
+		case '*':
+			if (needle[1] == '\0')
+				return true;
+			max = strlen(haystack);
+			for (i = 0; i < max; i++)
+				if (wild_match(haystack + i, needle + 1))
+					return true;
+			return false;
+		default:
+			if (toupper(*haystack) != *needle)
+				return false;
+			++haystack;
+		}
+	}
+	return *haystack == '\0';
```

Comment:
If `haystack` is null this *haystack will crash (even if the entire prior loop didn't run).

Reply:
Added assert for it. Done.